embassy-stm32/src/usart/ringbuffered.rs

```diff
@@ -243,11 +186,7 @@ impl<'d, T: BasicInstance, RxDma: super::RxDma<T>> RingBufferedUartRx<'d, T, RxD
 
         match select(dma, uart).await {
             Either::Left(((), _)) => Ok(()),
-            Either::Right((Ok(()), _)) => Ok(()),
-            Either::Right((Err(e), _)) => {
-                self.teardown_uart();
```

Comment:
Is it on purpose that the uart is not stopped on error?

Reply:
It's stopped in the read function.